src/main/java/main/EmployeesProcessor.java

```diff
@@ -0,0 +1,412 @@
+        switch (executionType){
+            case FOREACH:{
+                employeesByAddress = new HashSet<>();
+                for (Employee e:employeeList){
+                    if (e.getAddress().equals(address)){
+                        employeesByAddress.add(e.getFirstName());
+                    }
+                }
+                break;
+            }
+            case ITERATOR:{
+                employeesByAddress = new HashSet<>();
+                Iterator<Employee> employeeIterator = employeeList.iterator();
+                while (employeeIterator.hasNext()){
+                    Employee employee = employeeIterator.next();
+                    if (employee.getAddress().equals(address)){
+                        employeesByAddress.add(employee.getFirstName());
+                    }
+                }
+                break;
+            }
+            case STREAM:{
+                employeesByAddress = employeeList.stream()
+                        .filter(s->s.getAddress().equals(address))
+                        .map(s->s.getFirstName())
+                        .collect(Collectors.toSet());
+                break;
+            }
+        }
+        return employeesByAddress;
+    }
+
+    /**
+     * TASK #8
+     * Вивести дані просто кожного працівника в консоль і витягнути колекцію адрес
+     */
+    public Set<String> getAddressOfEmployees(List<Employee> employeeList, ExecutionType executionType){
+        Set<String> distinctAddressesOfEmployees = null;
+        switch (executionType){
+            case FOREACH:{
+                distinctAddressesOfEmployees = new HashSet<>();
+                for (Employee e:employeeList){
+                    distinctAddressesOfEmployees.add(e.getAddress());
+                }
+                break;
+            }
+            case ITERATOR:{
+                distinctAddressesOfEmployees = new HashSet<>();
+                Iterator<Employee> employeeIterator = employeeList.iterator();
+                while (employeeIterator.hasNext()){
+                    distinctAddressesOfEmployees.add(employeeIterator.next().getAddress());
+                }
+                break;
+            }
+            case STREAM:{
+                distinctAddressesOfEmployees = employeeList.stream()
+                        .map(s->s.getAddress())
+                        .collect(Collectors.toSet());
+                break;
+            }
+        }
+        return distinctAddressesOfEmployees;
+    }
+
+    /**
+     * TASK #9
+     * Витягнути колекцію імен працівників так, щоб нова колекція включала в себе 5 імен
+     */
+    public List<String> getFiveEmployeesName(List<Employee> employeeList, ExecutionType executionType){
+        List<String> fiveEmployeesName = null;
+        switch (executionType){
+            case FOREACH:{
+                fiveEmployeesName = new ArrayList<>();
+                for (Employee e:employeeList){
+                    if (!(fiveEmployeesName.size() == 5)) {
```

Comment:
In this case it will continue to iterate through collection even when we will already have required 5 elements.
If for ex this list will consist of 10000000000000000000 elements, it's execution will take some time
But we dont need it, we just need to get our 5 elements and go away from the method
Please think about how to fix it